File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    vector<int> q;
    for (int i = 0; i < score.size(); i++) {
        q.push_back(score[i]);
        sort(q.rbegin(), q.rend());
        if (q.size() < k) answer.push_back(q[q.size() - 1]);
        else answer.push_back(q[k - 1]);
    }
    return answer;
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    q = []
    for s in score :
        q.append(s)
        q.sort(reverse = True)
        if (len(q) < k) :
            answer.append(q[-1])
        else :
            answer.append(q[k - 1])
    return answer